src/datepicker/calendar.js

```diff
@@ -287,7 +287,7 @@ export default class Calendar extends React.Component<
         const activeElm = document.activeElement;
         // need to look for any tabindex >= 0 and ideally for not disabled
         // focusable by default elements like input, button, etc.
-        const focusable = this.root
+        const focusable = this.state.rootElement
```

Comment:
Does it need to be in the state? Also, the state type and initial value would need to be updated.

Reply:
I think it needs to be in state. If this was a ref, I would send the same ref to CalendarHeader. Initially the ref would be empty, and CalendarHeader would not render on the correct node. When the ref is set, it would not trigger a re-render and CalendarHeader would still be rendered on the wrong node. 

BUT, when I try to make this a ref, it still works. I think it works because other changes trigger re-renders after the ref has gotten the correct element, but I could be wrong. If that is the case, it seems like a fragile solution IMO. What do you think?